### DISCUSSION TITLE
Buffered `combineLatest`

### DISCUSSION THREAD

**Discussion by griest024** [ReactiveX/rxjs#7461] [Ideas / Feature request]
When multiple `combineLatest` sources emit at the same time (from an async perspective), I want the next emission of `combineLatest` to include the latest emissions from all of the sources. This is a pretty common use case in applications: where multiple subscribed pieces of state are updated from a single API call and a component will want to see updated data from all pieces of state (not just whichever one happens to win the race condition). Plus this will improve performance by reducing the rerenders from `n` pieces of subscribed state to just `1`.

So I wrote a [simple wrapper for `combineLatest` which accomplishes this](https://playcode.io/1820280). My question is, is there a creation function or operator already in rxjs that does this and if not, will you consider adding this feature?

**Comment by samal-rasmussen** [comment]
Your example has 3 sources which are combined with combineLatest, and then each source emits synchronously which results in three emits from the combineLatest with one value updating at a time:
1,0,0
1,1,0
1,1,1

Typical combineLatest.

Your fix buffers all synchronously emitted values by using a bufferTime(0), and then only emits the last value of all the accrued buffered values resulting in:
1,1,1

You solution looks clean enough. I don't think the maintainers are open to adding more functionality to rxjs, as they are happy that people can solve their own problems using the existing functionality.

What I am left wondering is that maybe what you want is signals and not an observable? Because you aren't interested in all emitted values, but only the end result which you probably want to show to the user. This is what signals are for, they are inherently lossy like this.

**Comment by hakimio** [comment]
You can just use `zip()` to achieve the same result.

**Reply by griest024** [reply]
The problem is that it will never emit `1,1,0` in the case where `c` doesn't emit for a while.

**Comment by AronHoxha** [comment]
Small note, you could use `debounceTime(0)` to replace the buffer, `filter` and `map` operators, or even use a different scheduler `debounceTime(0, asapScheduler)` to keep values in micro task queue

**Reply by samal-rasmussen** [reply]
You're right. You can replace the whole `collectLatest` helper with just `debounceTime(0)` and get the same output.
